multishard_mutation_query.cc

```diff
@@ -593,16 +593,15 @@ future<> read_context::save_readers(flat_mutation_reader_v2::tracked_buffer unco
     const auto cs_stats = dismantle_compaction_state(std::move(compaction_state));
     tracing::trace(_trace_state, "Dismantled compaction state: {}", cs_stats);
 
-    return do_with(std::move(last_pkey), std::move(last_ckey), [this] (const dht::decorated_key& last_pkey,
-            const std::optional<clustering_key_prefix>& last_ckey) {
-        return parallel_for_each(boost::irange(0u, smp::count), [this, &last_pkey, &last_ckey] (shard_id shard) {
+    co_await smp::invoke_on_all([this, &last_pkey, &last_ckey] {
```

Comment:
I don't see much value in coroutinizing this method as it is not helping readability. That said, this method is not on the hot-path (runs once per page), so I won't object the one added allocation. Converting this to coroutines helps uniformity (only performance critical methods remaining continuation based).

Reply:
Why added allocation?
There was a call to `do_with` before that is replaced by the coroutine frame allocation.

Reply:
Right, I missed that, so even better.